modules/seqexec/server/src/main/scala/seqexec/server/keywords/package.scala

```diff
@@ -3,84 +3,157 @@
   // because I cannot mix different types in a List. But at the end I only care about the value as a String, so I
   // use an internal representation, and offer a class to the developer (KeywordBag) to create the list from typed
   // keywords.
 
-  private[keywords] final case class InternalKeyword(name: String, keywordType: KeywordType, value: String)
+  private[keywords] final case class InternalKeyword(name: String,
+                                                     keywordType: KeywordType,
+                                                     value: String)
 
   final case class KeywordBag(keywords: List[InternalKeyword]) {
-    def add[T](k: Keyword[T]): KeywordBag = KeywordBag(keywords :+ internalKeywordConvert(k))
-    def append(other: KeywordBag): KeywordBag =  KeywordBag(keywords ::: other.keywords)
+
+    def add[T](k: Keyword[T]): KeywordBag =
+      KeywordBag(keywords :+ internalKeywordConvert(k))
+
+    def append(other: KeywordBag): KeywordBag =
+      KeywordBag(keywords ::: other.keywords)
   }
 
   //TODO: Add more apply methods if necessary
   @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
   object KeywordBag {
     def empty: KeywordBag = KeywordBag(List())
-    def apply[A](k1: Keyword[A]): KeywordBag = KeywordBag(List(internalKeywordConvert(k1)))
-    def apply[A, B](k1: Keyword[A], k2: Keyword[B]): KeywordBag = KeywordBag(List(internalKeywordConvert(k1), internalKeywordConvert(k2)))
-    def apply[A, B, C](k1: Keyword[A], k2: Keyword[B], k3: Keyword[C]): KeywordBag =
-      KeywordBag(List(internalKeywordConvert(k1), internalKeywordConvert(k2), internalKeywordConvert(k3)))
-    def apply[A, B, C, D](k1: Keyword[A], k2: Keyword[B], k3: Keyword[C], k4: Keyword[D]): KeywordBag =
-      KeywordBag(List(internalKeywordConvert(k1), internalKeywordConvert(k2), internalKeywordConvert(k3),
-        internalKeywordConvert(k4)))
-    def apply[A, B, C, D, E](k1: Keyword[A], k2: Keyword[B], k3: Keyword[C], k4: Keyword[D], k5: Keyword[E]): KeywordBag =
-      KeywordBag(List(internalKeywordConvert(k1), internalKeywordConvert(k2), internalKeywordConvert(k3),
-        internalKeywordConvert(k4), internalKeywordConvert(k5)))
-    def apply[A, B, C, D, E, F](k1: Keyword[A], k2: Keyword[B], k3: Keyword[C], k4: Keyword[D], k5: Keyword[E], k6: Keyword[F]): KeywordBag =
-      KeywordBag(List(internalKeywordConvert(k1), internalKeywordConvert(k2), internalKeywordConvert(k3),
-        internalKeywordConvert(k4), internalKeywordConvert(k5), internalKeywordConvert(k6)))
+
+    def apply[A](k1: Keyword[A]): KeywordBag =
+      KeywordBag(List(internalKeywordConvert(k1)))
+
+    def apply[A, B](k1: Keyword[A], k2: Keyword[B]): KeywordBag =
+      KeywordBag(List(internalKeywordConvert(k1), internalKeywordConvert(k2)))
+
+    def apply[A, B, C](k1: Keyword[A],
+                       k2: Keyword[B],
+                       k3: Keyword[C]): KeywordBag =
+      KeywordBag(
```

Comment:
Surely we can do better than all this copy/paste. How about

```scala
def apply(ks: Keyword[_]*): KeywordBag = 
  KeywordBag(ks.map(internalKeywordConvert).toList))
```

Or, better, `KeywordBag` is a monoid, which (when defined) would let you implement it as 

```scala
def apply(ks: Keyword[_]*): KeywordBag = 
  ks.toList.foldMap(KeywordBag.apply)
```

Reply:
`Keyword[_]` scares me 😄

Reply:
You can write it as `Keyword[A] forSome { type A }` if it makes you feel better ;-)

Reply:
My :+1: for the `Monoid` suggestion.